### PR TITLE
build(bzlmod): Switch to rules_oci

### PR DESCRIPTION
**Pull request by jasonschroeder-sfdc**
- remove `rules_docker` in favor of `rules_oci`
- add `[container_structure_test]()` in `//container/test/...` to ensure the output images:
  - have `.jar` in the right place
  - have execution wrappers in the right place
  - have a default minimal config, and a `CONFIG_PATH` env var to match
  - have a default Java logging config at `/app/build_buildfarm/src/main/java/build/buildfarm/logging.properties`
  - have telemetry tools in the expected place
  - DOES NOT TEST:
    - java jvm properties for Windows, telemetry

### CODE REVIEW

File: MODULE.bazel [hunk 1]
```diff
@@ -12,9 +12,37 @@ bazel_dep(name = "rules_cc", version = "0.0.9")
...
     name = "buildifier_prebuilt",
     version = "6.4.0",
     dev_dependency = True,
 )
+
+oci = use_extension("@rules_oci//oci:extensions.bzl", "oci")
+
+# Server base image
+oci.pull(
+    # Tnis is a multi-arch image!
```
Comment (werkt) [inline MODULE.bazel:28]: This == Tnis?

File: container/defs.bzl [hunk 2]
```diff
@@ -0,0 +1,28 @@
+"""Rules for ENV"""
+
+def _oci_image_env_impl(ctx):
+    """
+    Helper method to write out a "key=value" pair on sepaarate lines. This file is fed into oci_image() in the `env` kwargs.
```
Comment (werkt) [inline container/defs.bzl:5]: keep em sepaaaaaraaated.

File: container/test/worker_wrappers.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
+---
+schemaVersion: "2.0.0"
+# These align with the default paths in //src/main/java/build/buildfarm/common/config/ExecutionWrappers.java
+fileExistenceTests:
+  - name: "cgrousps cexec"
```
Comment (werkt) [inline container/test/worker_wrappers.yaml:5]: cgroups_exec typo

**Review by werkt — APPROVED** [review]
I think this is fine with a couple of typos

**Comment by jasonschroeder-sfdc** [comment]
Gotta fix the helm chart 🤦 before this is ready to merge.

**Pull request merged**